ratelearn/ratelearn/ratelearner.py

```diff
@@ -0,0 +1,132 @@
+import os
+import numpy as np
+import pandas as pd
+import torch
+import matplotlib.pyplot as plt
+from torch.utils.data import TensorDataset
+
+from ratelearn import RateMatrix, train_quantization
+
+
+class RateMatrixLearner:
+    def __init__(
+        self,
+        frequency_matrices: str,
+        output_dir: str,
+        stationnary_distribution: str,
+        mask: str = None,
+        frequency_matrices_sep="\s",
+        rate_matrix_parameterization="pande_reversible",
+    ):
+        self.frequency_matrices = frequency_matrices
+        self.frequency_matrices_sep = frequency_matrices_sep
+        self.output_dir = output_dir
+        self.stationnary_distribution = stationnary_distribution
+        self.mask = mask
+        self.rate_matrix_parameterization = rate_matrix_parameterization
+        self.lr = None
+        self.do_adam = None
+        self.df_res = None
+        self.Qfinal = None
+        self.trained_ = False
+
+        # Create experiment directory
+        if os.path.exists(self.output_dir):
+            raise ValueError("Please provide a nonexisting experiment path")
+        os.makedirs(self.output_dir)
+
+        # Open frequency matrices
+        self.quantized_data, self.n_states = self.get_branch_to_mat()
+
+        # Open stationnary distribution if necessary
+        pi_path = self.stationnary_distribution
+        if pi_path is not None:
+            pi = pd.read_csv(pi_path, header=None, index_col=None).values.squeeze()
+        else:
+            pi = np.ones(self.n_states)
+            pi = pi / pi.sum()
+        self.pi = torch.tensor(pi).float()
+
+        mask_path = self.mask
+        if mask_path is not None:
+            mask_mat = pd.read_csv(
+                mask_path, sep="\s", header=None, index_col=None
+            ).values
+        else:
+            mask_mat = np.ones((self.n_states, self.n_states))
+        self.mask_mat = torch.tensor(mask_mat)
+
+        self.mat_module = RateMatrix(
+            num_states=self.n_states,
+            mode=self.rate_matrix_parameterization,
+            pi=self.pi,
+            pi_requires_grad=True,
```

Comment:
@PierreBoyeau Shouldn't this depend on whether `pi` was provided by the user? So: `pi_requires_grad=pi_path is None`?

Reply:
My intuition was that even in the case where `pi_path` was provided, it would be seen as some sort of initialization based on observed data.
I think it is probably for the best to consider pi as a non-learnable parameter, but I am still curious to investigate the importance of this parameter.